### PR TITLE
[CST] Create new evidence submission record for each document upload

### PR DESCRIPTION
**Pull request by samcoforma**
We're creating an `evidence_submission` model and recording to it when evidence submission documents are uploaded in the Claim Status Tool. This will allow us to keep track of upload status so we prevent silent failures. We're keeping this work in the `feature/evidence-submissions-error-handling` feature branch for now since we'll be updating the model file and related migration a lot before it's ready to merge into `master`

## Summary

- *This work is behind a feature toggle (flipper): NO*
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- [[CST] Create a new record on evidence submission table every time a document is uploaded by a user](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94469)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by stevenjcumming** [comment]
@samcoforma I think if you update branch with master, it will resolve your security code check warnings

File: app/models/evidence_submission.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class EvidenceSubmission < ApplicationRecord
-  belongs_to :user_account
```
Comment (jerekshoe) [inline app/models/evidence_submission.rb:4]: Are we sure we want to remove this part?

File: db/migrate/20240925160219_create_evidence_submissions.rb [hunk 2]
```diff
@@ -4,10 +4,10 @@ def change
       t.string :job_id
       t.string :job_class
       t.string :claim_id
-      t.references :user_account, null: false, foreign_key: true, type: :uuid
```
Comment (jerekshoe) [inline db/migrate/20240925160219_create_evidence_submissions.rb:7]: This as well

**Comment by rmtolmach** [comment]
When tests and linting pass and the description is filled out, you can take this out of draft mode.

File: app/services/evss_claim_service.rb [hunk 3]
```diff
@@ -123,6 +124,20 @@ def record_workaround(task, claim_id, job_id)
                         })
   end
 
+  def record_evidence_submission(claim_id, job_id, tracked_item_id)
+    user_account_id = @user.user_account_uuid
+    user_account = UserAccount.find(@user.user_account_uuid)
+    job_class = self.class
+    upload_status = 'pending'
+    evidence_submission = EvidenceSubmission.build(claim_id:,
```
Comment (jerekshoe) [inline app/services/evss_claim_service.rb:132]: Is there a reason to use `.build` here instead of `.new`?

Comment (samcoforma) [reply app/services/evss_claim_service.rb:132]: @jerekshoe Yeah, I don't want it to save until after assigning the assocation. `.new` would save it right away.

File: db/migrate/20240925160219_create_evidence_submissions.rb [hunk 4]
```diff
@@ -5,9 +5,10 @@ def change
       t.string :job_class
       t.string :claim_id
       t.references :user_account, null: false, foreign_key: true, type: :uuid
+      # t.string :user_account_id
```
Comment (jerekshoe) [inline db/migrate/20240925160219_create_evidence_submissions.rb:8]: Assuming we want to remove this comment?

**Review by jerekshoe — APPROVED** [review]

**Pull request merged**